### PR TITLE
add function definintion to closured functions

### LINKED ISSUE #6857: Closure function naming
Currently returned closures in functions that require temp variables are unnanmed e.g.:

````javascript
THREE.Vector4.prototype = {

      clampScalar: ( function () {

		var min, max;

		return function ( minVal, maxVal ) {
		...
````
This means when debugging or outputting a stack trace on an error this is an anonymous function and will be reported as ```(anonymous function)```

Would it be worth naming these functions so they do get properly output? e.g.
````javascript
THREE.Vector4.prototype = {

      clampScalar: ( function () {

		var min, max;

		return function clampScalar( minVal, maxVal ) {
		...
````
Which will then be shown in the stack trace as ```clampScalar```

### PR DESCRIPTION
**Pull request by gero3**
fixes #6857

This add function definitions for (hopefully) all closured functions.

### CODE REVIEW

**Comment by benaadams** [comment]
That was fast O_o

**Comment by gero3** [comment]
I'm actually running through the issues and picking out the easy ones to do.

File: src/extras/core/Gyroscope.js [hunk 1]
```diff
@@ -1,65 +1,65 @@
-/**
```
Comment (mrdoob) [inline src/extras/core/Gyroscope.js]: Hmm, any idea why this file is marked as if all changed?

File: src/loaders/Loader.js [hunk 2]
```diff
@@ -1,398 +1,398 @@
...
-			}
-
-		}
-
-		return null;
-
-	}
-
-};
+/**
```
Comment (mrdoob) [inline src/loaders/Loader.js]: Also with this file...?

Comment (dubejf) [inline src/loaders/Loader.js]: Line ending crlf

Comment (gero3) [inline src/loaders/Loader.js]: Is my edit wrong or was the file in the repository wrong with line endings??

Comment (dubejf) [inline src/loaders/Loader.js]: Mixed line ending in the repository. This file uses CR-LF (Windows) line ending except for a single line that uses LF (Unix) that comes from a different commit.

I wonder if this repository should have a `.gitattributes`: https://help.github.com/articles/dealing-with-line-endings/

File: src/extras/core/Gyroscope.js (hunk 1 above)
Comment (gero3) [inline src/extras/core/Gyroscope.js]: Do I need to revert them??

because https://github.com/mrdoob/three.js/pull/6859?w=1 seems clean to me

**Comment by mrdoob** [comment]
I've downloaded https://github.com/gero3/three.js/archive/addFunctionDefintions.zip, tried running `utils/build/build.sh` and I can't find "function clampScalar" in `three.min.js`. I think the compiler is removing them 😕

**Comment by gero3** [comment]
> I've downloaded https://github.com/gero3/three.js/archive/addFunctionDefintions.zip, tried running utils/build/build.sh and I can't find "function clampScalar" in three.min.js. I think the compiler is removing them 😕

That's somewhat a pity but i agree actually with @benaadams that a non minified version is better for looking up function names anyways.

**Comment by benaadams** [comment]
> I've downloaded https://github.com/gero3/three.js/archive/addFunctionDefintions.zip, tried running utils/build/build.sh and I can't find "function clampScalar" in three.min.js. I think the compiler is removing them

Best of both worlds then! Good annotations for debugging; still small minified size :wink:

**Comment by mrdoob** [comment]
Oh! Now I get it 😀

**Comment by mrdoob** [comment]
Merged. Thanks thanks!

**Pull request merged**